chart_review/agree.py

```diff
@@ -178,3 +178,59 @@ def csv_row_score(
 
     row.append(pick_label if pick_label else "*")
     return "\t".join(row)
+
+
+def contingency_table(
+    annotations: defines.ProjectAnnotations,
+    truth: str,
+    annotator1: str,
+    annotator2: str,
+    note_range: Collection[int],
+    labels: Optional[Iterable[str]] = None,
+) -> dict[str, list]:
+    # Grab all mentions
+    mentions = {
+        annotator: annotations.mentions.get(annotator, defines.Mentions())
+        for annotator in {truth, annotator1, annotator2}
+    }
+
+    # Only examine labels that were used by any compared annotators at least once
+    label_set = set()
+    for mentions_set in mentions.values():
+        for mention_labels in mentions_set.values():
+            label_set |= set(mention_labels)
+    if labels:
+        label_set &= set(labels)
+
+    BC = []  # both correct
+    OL = []  # only left
+    OR = []  # only right
+    BW = []  # both wrong
```

Comment:
suggestion: just looking at the usual language around this (i.e. positive/negative) - Both positive/left positive/right positive/both negative? Or PP/PN/NP/NN?

Reply:
I did just make these terms up, I am not wedded to them. I think McNemar uses a/b/c/d which... I hate.

But using Pos/Neg here risks getting confused with the actual clinical answer. e.g. both annotators answering negative could mean both are correct (if truth said negative too) or both are wrong (if truth said positive).

But @Dtphelan1 - comment here on that and/or on better wording in general.

Reply:
I lean towards your intuition @mikix, as evidence by my previous conflation of Contingency Tables and Confusion Matrices. I'm lean towards these names, but it's a light inclination.